File: Java/1281_Subtract_the_Product_and_Sum_of_Digits_of_an_Integer.java
class Solution {
    public int subtractProductAndSum(int n) {
        int productOfDigits = 1;
        int sumOfDigits = 0;
        while(n > 0){
            int digit = n%10;
            productOfDigits *= digit;
            sumOfDigits += digit;
            n /= 10;
        }
        return productOfDigits - sumOfDigits;
    }
}

File: Cpp/1281_Subtract_the_Product_and_Sum_of_Digits_of_an_Integer.cpp
class Solution {
public:
    int subtractProductAndSum(int n) {
        int p = 1;
        int s = 0;
        int temp;
        while(n > 0){
            temp = n%10;
            p *= temp;
            s += temp;
            n /= 10;
        }
        return p - s;
    }
};
